rider/src/main/kotlin/com/jetbrains/rider/plugins/godot/MetadataCoreFileWatcher.kt

```diff
@@ -40,15 +40,17 @@ class MetadataCoreFileWatcher(project: Project) : LifetimedProjectComponent(proj
     }
 
     init {
-        project.solution.isLoaded.whenTrue(componentLifetime) {l->
+        project.solution.isLoaded.whenTrue(componentLifetime) { l ->
             val godotDiscoverer = GodotProjectDiscoverer.getInstance(project)
             godotDiscoverer.isGodotProject.whenTrue(l) { lt ->
-                thread(name = "MetadataFileWatcher") {
+                lt.launchBackground {
                     val watchService: WatchService = FileSystems.getDefault().newWatchService()
                     val metaFileDir = project.solutionDirectory.resolve(cfgDir).toPath()
 
-                    if (!(metaFileDir.isDirectory()))
-                        return@thread
+                    while (!(metaFileDir.isDirectory())) {
+                        // wait for folder to appear
+                        delay(1000)
```

Comment:
Maybe add some reasonable timeout on this process, after that we'll give up? Or no such timeout exists?